### ISSUE TITLE
Using the emitter 

### ISSUE THREAD

**Issue by RichardLitt** [philschatz/octokat.js#109]
I am continually running into rate limits for the GitHub API, and it's pretty hard for me to diagnose my issues easily. In part, this is because [my code](//github.com/RichardLitt/name-your-contributors) has hundred of API calls, so it makes sense. I'd like to have a better picture into my rate limits as they happen by console logging what requests I am making and what rate limits I have left, after each one. 

Someone pointed me to [this line](https://github.com/philschatz/octokat.js/blob/111cd9386c3f6342a72d64b7ad335efa8f504f37/src/request.coffee#L157) in the code. Is there an easy way for me to use this? 

Thanks.

**Comment by philschatz** [comment]
Yep. It's not "officially" supported but you should be able to do something like the following:

```
options = {
  emitter: {
    emit: function(eventType, method, path, data, options) { 
      console.log(arguments); 
    }
  }
};
octo = require('octokat')(options)
octo.repos('philschatz/octokat.js').fetch()
```

You can try it out by changing the first line at https://tonicdev.com/npm/octokat

Hope that helps!
